### ISSUE TITLE
Warn/error if subprocess command returns nonzero code

### ISSUE THREAD

**Issue by bdice** [glotzerlab/signac-flow#170]
### Feature description

Currently Python operations that raise an error will halt the execution loop of `project.py run`. However, the loop doesn't halt if the erroring operation is forked via `@directives(fork=True)` or `@cmd`. Non-zero exit codes from the `subprocess.call` are silently ignored.

### Proposed solution

Non-zero exit codes should be caught, and either throw a warning or raise an error.

### Additional context

Sample project file:

```python
from flow import FlowProject, cmd, directives

class Project(FlowProject):
    pass

@Project.operation
@directives(fork=True)
def raise_error(job):
    raise RuntimeError('there is no spoon, only this fork')

@Project.operation
@cmd
def exit_1(job):
    return 'exit 1'

if __name__ == '__main__':
    Project().main()
```

Sample output:
- the repeated `UnknownEnvironment` lines are from the forked Python processes
- `--show-traceback` is ignored by the forked Python processes (but may be resolved separately by #169)
- the operation `exit 1` runs but its errors are silently ignored

```python
user@HOST ~/tmp_signac $ python project.py run --show-traceback
Using environment configuration: UnknownEnvironment
Using environment configuration: UnknownEnvironment
ERROR: Encountered error during program execution: 'there is no spoon, only this fork'
Execute with '--show-traceback' or '--debug' to get more information.
Using environment configuration: UnknownEnvironment
ERROR: Encountered error during program execution: 'there is no spoon, only this fork'
Execute with '--show-traceback' or '--debug' to get more information.
Using environment configuration: UnknownEnvironment
ERROR: Encountered error during program execution: 'there is no spoon, only this fork'
Execute with '--show-traceback' or '--debug' to get more information.
WARNING:flow.project:Operation 'raise_error' has no post-conditions!
WARNING:flow.project:Operation 'exit_1' has no post-conditions!
```

Sample of what it looks like to warn on non-zero exit codes:
```python
user@HOST ~/tmp_signac $ python project.py run --show-traceback
Using environment configuration: UnknownEnvironment
Using environment configuration: UnknownEnvironment
ERROR: Encountered error during program execution: 'there is no spoon, only this fork'
Execute with '--show-traceback' or '--debug' to get more information.
WARNING:flow.project:Operation 'raise_error(1df6f548394a0d3ff0dffc45721bd1df)' called via subprocess returned nonzero code: 1.
WARNING:flow.project:Operation 'exit_1(1df6f548394a0d3ff0dffc45721bd1df)' called via subprocess returned nonzero code: 1.
Using environment configuration: UnknownEnvironment
ERROR: Encountered error during program execution: 'there is no spoon, only this fork'
Execute with '--show-traceback' or '--debug' to get more information.
WARNING:flow.project:Operation 'raise_error(d9a61015c4b65297ce8b9065a3329dc3)' called via subprocess returned nonzero code: 1.
WARNING:flow.project:Operation 'exit_1(d9a61015c4b65297ce8b9065a3329dc3)' called via subprocess returned nonzero code: 1.
Using environment configuration: UnknownEnvironment
ERROR: Encountered error during program execution: 'there is no spoon, only this fork'
Execute with '--show-traceback' or '--debug' to get more information.
WARNING:flow.project:Operation 'raise_error(f002b179c4eaac8e1113f56d803c8d51)' called via subprocess returned nonzero code: 1.
WARNING:flow.project:Operation 'exit_1(f002b179c4eaac8e1113f56d803c8d51)' called via subprocess returned nonzero code: 1.
WARNING:flow.project:Operation 'exit_1' has no post-conditions!
WARNING:flow.project:Operation 'raise_error' has no post-conditions!
```

**Comment by csadorf** [comment]
I believe that switching to using `subprocess.run()` with `check=True` and then propagating the `CalledProcessError` makes the most sense. If users want to use applications that exit with non-zero code in a non-error state than they need to wrap those themselves.

**Comment by vyasr** [comment]
Resolved in #172
